app/services/device_notifier.rb

```diff
@@ -0,0 +1,77 @@
+class DeviceNotifier
+
+  # Used for testing notifications without talking to Apple services.
+  # Intended to be compatible with Houston::Client
+  class FakeAPNSClient
+    attr_reader :delivered_notifications
+
+    def initialize
+      @delivered_notifications = []
+    end
+
+    def push(payload)
+      @delivered_notifications << payload
+      Rails.logger.debug("Pushed fake notification_payload: #{payload.inspect}")
+    end
+  end
+
+  def self.notification_environment=(val)
+    new_environment = ActiveSupport::StringInquirer.new(val.to_s)
+    if new_environment.production? || new_environment.development? || new_environment.fake?
+      @@notification_environment = new_environment
+    else
+      raise ArgumentError.new("unknown notification environment: #{new_environment}")
+    end
+  end
+
+  def self.notification_environment
+    @@notification_environment
+  end
+
+  # It's important to regularly remove devices that Apple reports as stale. If
+  # you continue to try to notify stale devices, Apple may revoke your access
+  # to APNS!
+  def deactivate_stale_devices!
```

Comment:
I'm peering at the apns from the outside here, so forgive the obvious questions.

When would you run this rake task? I was imagining that the user announces their device token, and old tokens would be removed at that time (up in the devices controller).